packages/react-router-dom/dom.ts

```diff
@@ -127,12 +127,22 @@ export type SubmitTarget =
   | null;
 
 // One-time check for submitter support
-let formDataSupportsSubmitter = false;
-try {
-  // @ts-expect-error if FormData supports the submitter parameter, this will throw
-  new FormData(undefined, 0);
-} catch (e) {
-  formDataSupportsSubmitter = true;
+let _formDataSupportsSubmitter: boolean | null = null;
+
+function isFormDataSubmitterSupported() {
+  if (_formDataSupportsSubmitter === null) {
+    try {
+      new FormData(
+        document.createElement("form"),
+        // @ts-expect-error if FormData supports the submitter parameter, this will throw
+        0
+      );
+      _formDataSupportsSubmitter = false;
+    } catch (e) {
+      _formDataSupportsSubmitter = true;
+    }
+  }
+  return _formDataSupportsSubmitter;
```

Comment:
Moved this to a function so it wouldn't try to execute in the module scope on the server and `document` wasn't defined